packages/firebase_core/tizen/src/firebase_core_tizen_plugin.cc

```diff
@@ -118,33 +131,29 @@ void FirebaseCoreTizenPlugin::InitializeApp(
   firebase::App* app = firebase::App::GetInstance(app_name.c_str());
   if (!app) {
     app = firebase::App::Create(
-        PigeonOptionsToFirebaseOptions(initialize_app_request),
+        PigeonFirebaseOptionsToAppOptions(initialize_app_request),
         app_name.c_str());
   }
 
-  PigeonInitializeResponse message_response;
-  message_response.set_name(app->name());
-  message_response.set_options(FirebaseOptionsToPigeonOptions(app->options()));
-
-  result(message_response);
+  // Send back the result to Flutter
+  result(AppToPigeonInitializeResponse(*app));
 }
 
 void FirebaseCoreTizenPlugin::InitializeCore(
     std::function<void(ErrorOr<flutter::EncodableList> reply)> result) {
-  flutter::EncodableList response;
+  // TODO: Missing function to get the list of currently initialized apps
+  std::vector<PigeonInitializeResponse> initializedApps;
+  std::vector<App*> all_apps = App::GetApps();
+  for (const App* app : all_apps) {
+    initializedApps.push_back(AppToPigeonInitializeResponse(*app));
+  }
 
-  std::vector<firebase::App*> firebase_apps = firebase::App::GetApps();
-  for (auto app : firebase_apps) {
-    PigeonInitializeResponse message_response;
-    message_response.set_name(app->name());
-    message_response.set_options(
-        FirebaseOptionsToPigeonOptions(app->options()));
+  flutter::EncodableList encodableList;
 
-    response.push_back(
-        flutter::EncodableValue(message_response.ToEncodableMap()));
+  for (const auto& item : initializedApps) {
+    encodableList.push_back(flutter::CustomEncodableValue(item));
   }
-
-  result(response);
+  result(encodableList);
```

Comment:
I think it can be made as one loop.

ex) ?
```
flutter::EncodableList encodableList;
std::vector<App*> all_apps = App::GetApps();
  for (const App* app : all_apps) {
   encodableList.push_back(flutter::CustomEncodableValue(AppToPigeonInitializeResponse(*app)));
  }
```